go/tools/builders/generate_test_main.go

```diff
@@ -228,20 +237,12 @@ func main() {
 	panicOnExit0Flag.Set("true")
 {{end}}
 {{if ne .CoverMode ""}}
-	if len(coverdata.Counters) > 0 {
-		testing.RegisterCover(testing.Cover{
-			Mode: "{{ .CoverMode }}",
-			Counters: coverdata.Counters,
-			Blocks: coverdata.Blocks,
-		})
-
-		if coverageDat, ok := os.LookupEnv("COVERAGE_OUTPUT_FILE"); ok {
-			{{if eq .CoverFormat "lcov"}}
-			flag.Lookup("test.coverprofile").Value.Set(coverageDat+".cover")
-			{{else}}
-			flag.Lookup("test.coverprofile").Value.Set(coverageDat)
-			{{end}}
-		}
+	if coverageDat, ok := os.LookupEnv("COVERAGE_OUTPUT_FILE"); ok {
```

Comment:
The logic above depends on the ugly panic on exit 0 hook. If I remember correctly, the coverage redesign introduced a more principled exit hook approach. Do you happen to know how this works?

I am asking because using the new technique would potentially allow us to Instrument non-test binaries and capture coverage information for them. Is that something you are also interested in?

Reply:
I've updated the code to drop a dependency on the panic on exit hook. I've also filed https://github.com/golang/go/issues/74749 to see if we can remove all logic to augment coverage source files.